internal/api/upload.go

```diff
@@ -0,0 +1,82 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"net/http"
+	"os"
+)
+
+func (c *ApiClient) Upload(ctx Context, reportUploadType string, uploadDate string, email string, file *os.File) error {
+	var body bytes.Buffer
+	var uploadDateTransformed *model.Date
+	var req *http.Request
+
+	if uploadDate != "" {
+		uploadDateFormatted := model.NewDate(uploadDate)
+		uploadDateTransformed = &uploadDateFormatted
+	}
+
+	err := json.NewEncoder(&body).Encode(model.Upload{
+		ReportUploadType: reportUploadType,
+		UploadDate:       uploadDateTransformed,
+		Email:            email,
+		File:             file,
```

Comment:
does this actually work with `*os.File`? I'm guessing really this is expecting some base64 version of the contents, and then you'd be able to change the `file` param above to `io.Reader` or some simpler type